hazelcast/src/main/java/com/hazelcast/internal/util/ExceptionUtil.java

```diff
@@ -297,7 +297,12 @@ MethodType signature() {
             <T extends Throwable> T cloneWith(MethodHandle constructor, String message,
                                               @Nullable Throwable cause) throws Throwable {
                 T cloned = (T) constructor.invokeWithArguments(message);
-                cloned.initCause(cause);
+                try {
+                    cloned.initCause(cause);
+                } catch (IllegalStateException ignored) {
+                    // Cause can be already set by the exception. It can be set to null as well.
```

Comment:
Not being able to tell the difference between if the cause is not set by the constructor or set to null is a bummer. I didn't like using exception handling for this scenario, but it seems like this is our only option.

Reply:
me neither.